core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -185,6 +364,24 @@ struct HIPParallelLaunch<
   using base_t = HIPParallelLaunchKernelInvoker<
       DriverType, Kokkos::LaunchBounds<MaxThreadsPerBlock, MinBlocksPerSM>,
       HIPLaunchMechanism::LocalMemory>;
+#else
+template <typename DriverType, typename LaunchBounds = Kokkos::LaunchBounds<>,
+          HIPLaunchMechanism LaunchMechanism =
+              DeduceHIPLaunchMechanism<DriverType>::launch_mechanism>
```

Comment:
This is the only line that really differs between these two; the rest is essentially the same. I see no reason to duplicate this much code; just put the preprocessor `#if` around this line and the corresponding line in the above `#if` block.